app/src/main/java/com/amaze/filemanager/filesystem/files/FileUtils.java

```diff
@@ -389,26 +389,29 @@ protected void onPostExecute(String s) {
     }.execute(paths);
   }
 
-  public static void shareFiles(ArrayList<File> a, Activity c, AppTheme appTheme, int fab_skin) {
+  public static void shareFiles(
+      ArrayList<File> files, Activity activity, AppTheme appTheme, int fab_skin) {
 
     ArrayList<Uri> uris = new ArrayList<>();
-    boolean b = true;
-    for (File f : a) {
-      uris.add(FileProvider.getUriForFile(c, c.getPackageName(), f));
-    }
+    boolean isGenericFileType = false;
```

Comment:
please verify the logic here, just to making sure something is not broken whilst refractoring.